app/modules_test.go

```diff
@@ -0,0 +1,75 @@
+package app_test
+
+import (
+	"fmt"
+	"time"
+	"testing"
+
+
+
+    "github.com/stretchr/testify/suite"
+	"github.com/regen-network/regen-ledger/types/testutil/network"
+	//data "github.com/regen-network/regen-ledger/x/data/client/testsuite"
+	group "github.com/regen-network/regen-ledger/x/data/group/testsuite"
+	"github.com/regen-network/regen-ledger/app"	
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	dbm "github.com/tendermint/tm-db"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+
+)
+// NewSimApp is not used in repository, why we need it?
+func NewSimApp(val network.Validator) servertypes.Application {
```

Comment:
I suggest to remove val argument here. SimApp is called within DefaultConfig only.

Reply:
We cannot, because of the `AppConstructor` signature.

Also, let's not use simapp anymore

```suggestion
func NewRegenAppConstructor(val network.Validator) servertypes.Application {
```